crates/testlang/outputs/cargo/tests/src/query/parser_tests.rs

```diff
@@ -52,12 +52,45 @@ fn test_zero_or_more_canonicalisation() {
 // Test the error message on parse failure
 #[test]
 fn test_parsing_error() {
-    let result = Query::parse(r#"@root [_ ..."#);
+    let result = Query::parse(r#"@root [_"#);
     match result {
-        Ok(_) => panic!("Expected error"),
+        Ok(_) => panic!("Expected parse failure"),
+        Err(e) => assert_eq!(e.message, "Parse error:\nexpected ']' at: \nAlt at: [_\n"),
+    }
+}
+
+#[test]
+fn test_fails_parsing_ellipsis() {
+    let result = Query::parse(r#"[_ ...]"#);
```

Comment:
IIRC, we discussed still parsing ellipsis, but producing a specific error message about it being deprecated, and how it was replaced with anchors, and how to use that instead. Are we still planning on doing that?

Reply:
Given it's early days, my €0.02 would be to not have 'legacy' stuff.

Reply:
A few people are already using queries and giving feedback about it. I think the slightly improved message would make it easier for them to upgrade here. Maybe we can remove this warning before v1? I think we will have a few releases before that.

Reply:
Yes, you're right. I forgot about it. I'll add a custom message for that case.

Reply:
Turns out it's not easy to add custom error messages to `nom` (I also see #845 open) unless we use a custom error type and propagate it all the way up. For now, I added a generic `Fail` error providing additional information using `context`.

Reply:
The changes to support this are contained, so it should be easy to remove them before v1 if necessary.